src/main/java/com/thoughtworks/aceleradora/controladores/MinhaListaControlador.java

```diff
@@ -10,12 +11,17 @@
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 
+import java.util.function.Consumer;
+
 @Controller
 @RequestMapping("/minha-lista")
 public class MinhaListaControlador {
 
     private ProdutoServico produtoServico;
     private MinhaListaServico minhaListaServico;
+    private final Consumer<Breadcrumb> partesComunsDoBreadCrumb = breadcrumb -> breadcrumb
```

Comment:
Não dava pra extrair o método que pega as páginas em comum pra dentro do próprio componente do `Breadcrumb`?

Reply:
Sim, daria. Mas é válido lembrar que `breadcrumb` é um objeto em escopo de requisição e será fornecido em branco pelos metódos das controladoras cada vez que isso seja necessário, logo, a cada requisição o objeto breadcrumb seria populado de maneira isolada e com as páginas necessárias para aquela rota, logo, não haveria duplicação de páginas, teoricamente. Poderíamos mudar a lista de páginas interna por um `Set` para prevenir possíveis duplicações.

Este modelo baseado em `Consumer` foca mais em evitar duplicação de códigos do que de dados, de fato.

Outra alternativa poderia ser o uso de herança ou algo semelhante:

```java
class MinhaListaBreadcrumb extends Breadcrumb {
  MinhaListaBreadcrumb() {
    super(new Pagina("Inicio", "/"), new Pagina(...) ...);
  }
}
```

Mas pareceu um tanto "overkill" para esse problema nesse momento.

Reply:
Outra forma, que poderia ser interessante é ter métodos de atalho:

```java
class Breadcrumb {
  // ... 
  public Breadcrumb minhaLista() {
    pagina("Inicio", "/").pagina("Listas", "/minha-lista");
    return this;
  }
}
```

Eu preferi usar o `Consumer` em vez desses métodos de atalho por conta do acoplamento. Os métodos de atalho forçam a classe `Breadcrumb` a conhecer detalhes de como os controllers e rotas se organizam. Com `Consumer`, essa imposição deixa de existir e passa a tornar-se responsabilidade do `Controller`, o que parece mais coerente.